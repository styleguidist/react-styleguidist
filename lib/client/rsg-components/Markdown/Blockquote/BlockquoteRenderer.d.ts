import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
interface BlockquoteProps extends JssInjectedProps {
    children: React.ReactNode;
    className?: string;
}
export declare const BlockquoteRenderer: React.FunctionComponent<BlockquoteProps>;
declare const _default: React.ComponentType<Pick<BlockquoteProps, "children" | "className">>;
export default _default;
