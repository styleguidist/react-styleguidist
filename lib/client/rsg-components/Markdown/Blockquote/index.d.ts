export { default } from 'rsg-components/Markdown/Blockquote/BlockquoteRenderer';
