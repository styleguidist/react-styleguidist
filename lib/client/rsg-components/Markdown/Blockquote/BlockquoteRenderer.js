import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var space = _ref.space,
      color = _ref.color,
      fontSize = _ref.fontSize,
      fontFamily = _ref.fontFamily;
  return {
    blockquote: {
      margin: [[space[2], space[4]]],
      padding: 0,
      color: color.base,
      fontFamily: fontFamily.base,
      fontSize: fontSize.base,
      lineHeight: 1.5
    }
  };
};

export var BlockquoteRenderer = function BlockquoteRenderer(_ref2) {
  var classes = _ref2.classes,
      className = _ref2.className,
      children = _ref2.children;
  var blockquoteClasses = cx(classes.blockquote, className);
  return /*#__PURE__*/React.createElement("blockquote", {
    className: blockquoteClasses
  }, children);
};
BlockquoteRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
export default Styled(styles)(BlockquoteRenderer);