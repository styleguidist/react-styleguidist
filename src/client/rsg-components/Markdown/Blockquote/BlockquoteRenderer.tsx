import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../../typings';

const styles = ({ space, color, fontSize, fontFamily }: Rsg.Theme) => ({
	blockquote: {
		margin: [[space[2], space[4]]],
		padding: 0,
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		lineHeight: 1.5,
	},
});

interface BlockquoteProps extends JssInjectedProps {
	children: React.ReactNode;
	className?: string;
}

export const BlockquoteRenderer: React.FunctionComponent<BlockquoteProps> = ({
	classes,
	className,
	children,
}) => {
	const blockquoteClasses = cx(classes.blockquote, className);
	return <blockquote className={blockquoteClasses}>{children}</blockquote>;
};

BlockquoteRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	className: PropTypes.string,
	children: PropTypes.any.isRequired,
};

export default Styled<BlockquoteProps>(styles)(BlockquoteRenderer);
