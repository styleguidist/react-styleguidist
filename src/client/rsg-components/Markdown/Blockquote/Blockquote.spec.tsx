import React from 'react';
import renderer from 'react-test-renderer';
import Blockquote from './index';

describe('Markdown Blockquote', () => {
	it('should render a blockquote', () => {
		const actual = renderer.create(
			<Blockquote>To be, or not to be: that is the question</Blockquote>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should preserve custom css class', () => {
		const actual = renderer.create(
			<Blockquote className="test-class">To be, or not to be: that is the question</Blockquote>
		);

		expect(actual).toMatchSnapshot();
	});
});
