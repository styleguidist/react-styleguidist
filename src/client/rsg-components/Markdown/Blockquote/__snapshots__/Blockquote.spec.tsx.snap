// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Blockquote should preserve custom css class 1`] = `
<blockquote
  className="rsg--blockquote-0 test-class"
>
  To be, or not to be: that is the question
</blockquote>
`;

exports[`Markdown Blockquote should render a blockquote 1`] = `
<blockquote
  className="rsg--blockquote-0"
>
  To be, or not to be: that is the question
</blockquote>
`;
